packages/core/src/application.ts

```diff
@@ -216,6 +222,26 @@ export class Application extends Context {
   public setMetadata(metadata: ApplicationMetadata) {
     this.bind(CoreBindings.APPLICATION_METADATA).to(metadata);
   }
+
+  /**
+   * Register a life cycle observer class
+   * @param ctor A class implements LifeCycleObserver
+   * @param name Optional name for the life cycle observer
+   */
+  public lifeCycleObserver<T extends LifeCycleObserver>(
+    ctor: Constructor<T>,
+    name?: string,
+  ): Binding<T> {
+    debug('Adding life cycle observer %s', name || ctor.name);
+    const binding = createBindingFromClass(ctor, {
+      name,
+      namespace: CoreBindings.LIFE_CYCLE_OBSERVERS,
+      type: CoreTags.LIFE_CYCLE_OBSERVER,
+      defaultScope: BindingScope.SINGLETON,
+    }).tag(CoreTags.LIFE_CYCLE_OBSERVER);
```

Comment:
Can we use `.apply(asLifeCycleObserver)` here? Would the following work?

```suggestion
    }).apply(asLifeCycleObserver)
```